src/Tokenizer/Tokens.php

```diff
@@ -487,7 +487,31 @@ public function findGivenKind($possibleKind, int $start = 0, ?int $end = null):
             }
         }
 
-        return \is_array($possibleKind) ? $elements : $elements[$possibleKind];
+        return $elements;
+    }
+
+    /**
+     * @param int|non-empty-list<int> $possibleKind kind or array of kinds
```

Comment:
```suggestion
     * @param int $possibleKind
```
to enforce the correct method softly

Reply:
not on v3, where both calls are allowed, but one is deprecated.
it can blow-up SCA analysis for repos of custom fixers.

Reply:
> it can blow-up SCA analysis for repos of custom fixers.

Why is that a problem? It should be actually desired, as static analysis will tell the developers to use the other method, but the fixers will still run without issues as before.

Reply:
IMHO, when introducing such change, we shouldn't fail someone CI, but we should give them deprecation notice.
If they use PHPStan, this could break their builds on Fixer non-major upgrade. With usage of PHPStan, param/return types can be consider as BC contract.

Reply:
btw, consider to give opinion on RFC itself, not detail of phpdoc annotation.